### PR TITLE
[FE] [ADD] Hamburger navigation for visitor & resto Web

### PR DESCRIPTION
**Pull request by RenanDbs**
### 🛠 Changes being made

Addition of the hamburger navigation for visitor & resto Web
So it's easyer to navigate on both website 

### ✨ What's the context?

What's the context for the changes? Are there any

### 🧠 Rationale behind the change

Why did you choose to make these changes? Were there any trade-offs you had to consider?

### 🧪 Test plan

How do you know the changes are safe to ship to production?

### 📸 Screenshots (optional)

If you made UI changes, what are the before an afters?

### 🏎 PR checklist

1. Read the ticket
- [ ] Read the related ticket on Jira and understand the requirements for this ticket. In case anything is unclear, ask the ticket creator or ticket executor.
2. Matching PR description
- [ ] Check that the PR description on GitHub is aligned with the ticket description on Jira.
3. Review the changes on GitHub
- [ ] is the current version of the destination branch (mostly main) merged into the PR branch?
- [ ] does it fulfill the definition of done for the ticket?
- [ ] were the tests adjusted or extended for the changed behavior?
- [ ] was the documentation adjusted or extended for changed behavior?
- [ ] does it align with the team code guidelines?
- [ ] check readability of code - is it easy to understand what is happening in the code snippets?
- [ ] proper naming
- [ ] is it logical?
- [ ] are there any performance issues?
- [ ] could it be improved / simplified?
4. Execute locally / in testing environment
- [ ] Execute the code locally and check if there are any issues. Test whether the new behavior from the ticket acts as expected. Try to think of edge cases and test them.
5. Run tests
- [ ] Run the tests locally and check that all of them succeed.
6. Linting & Formatting
- [ ] Check that the changes didn’t cause any linting or formatting warnings of errors.
7. Submit Feedback
- [ ] Approve if all of the following points fulfilled:
   - all previous steps did not reveal any issues
   - there are no open comments on the PR

- [ ] otherwise, request changes with specific comments (constructive) on what to change!

### CODE REVIEW

File: packages/restoWeb/src/components/dumpComponents/Header/Header.tsx [hunk 1]
```diff
@@ -53,50 +56,37 @@ const Header = () => {
...
   return (
+    <div>
+    <Drawer anchor="left" open={showDrawer} onClose={() => setShowDrawer(false)} classes={{ paper: styles.drawer }}>
+      <div className={styles.drawerContent}>
+        {!loggedIn ? (
+          <span className={styles.NavTitle} onClick={() => navigate('/login')}>{t('components.Header.login')}</span>
+        ) : (
+          <>
+            <span className={styles.NavTitle} onClick={logoutUser}>{t('components.Header.logout')}</span>
+            <span className={styles.NavTitle} onClick={() => navigate('/')} >{t('components.Header.home')}</span>
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/dumpComponents/Header/Header.tsx:71]: please use href for logout and home as well. it doesnt close hamburger menu for me after redirecting there otherwise (when using navigate()

File: packages/visitorWeb/src/components/Header/Header.tsx [hunk 2]
```diff
@@ -97,9 +91,30 @@ const Header = () => {
...
+              <a onClick={logoutUser}>
+                {t('components.Header.logout')}
+              </a>
+            ) : (
+              <a onClick={() => NavigateTo('/login', navigate, {})}>
+                {t('components.Header.login')}
+              </a>
+            )}
+          </span>
+          { loggedIn && (
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/Header/Header.tsx:111]: home (search screen) should be added in the navigation as well (in the topbar menu it was there with the guardos icon)

File: packages/restoWeb/src/components/dumpComponents/Header/Header.tsx [hunk 3]
```diff
@@ -53,50 +56,37 @@ const Header = () => {
   const changeLanguage = (language: string) => {
     i18n.changeLanguage(language);
     setShowLanguageDropdown(false);
+    setShowDrawer(false);
   };
 
   return (
+    <div>
+    <Drawer anchor="left" open={showDrawer} onClose={() => setShowDrawer(false)} classes={{ paper: styles.drawer }}>
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/dumpComponents/Header/Header.tsx:64]: when hovering over one submenu, the white line appears at the very bottom of the hamburger menu instead of underlining the submenu point. works however in visitor web

File: packages/visitorWeb/src/components/Header/Header.tsx [hunk 4]
```diff
@@ -57,33 +60,24 @@ const Header = () => {
...
       <div className={styles.header}>
+        <a
+          className={styles.NavTitle}
+          onClick={() => {
+            setShowNavigationDrawer(!showNavigationDrawer);
+          }}
+        >
+          <MenuIcon fontSize="large" style={{ color: 'white' }} />
+        </a>
         <div className={styles.logoContainer} onClick={() => NavigateTo('/', navigate, {})}>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/Header/Header.tsx:71]: should only navigate to home if not logged in

File: packages/restoWeb/src/components/dumpComponents/Header/Header.tsx [hunk 5]
```diff
@@ -53,50 +56,37 @@ const Header = () => {
...
+      </div>
+    </Drawer>
     <div className={styles.containerHeader}>
       <div className={styles.header}>
-        <div className={styles.logoContainer} onClick={() => NavigateTo('/', navigate, {})}>
+        <div className={styles.menuIcon} onClick={() => setShowDrawer(true)}>
+          <MenuIcon fontSize="large" style={{ color: 'white' }} />
+        </div>
+        <div className={styles.logoContainer} onClick={() => navigate('/')}>
           <div className={styles.logo}></div>
```
Comment (JosefineM2000) [inline packages/restoWeb/src/components/dumpComponents/Header/Header.tsx:87]: logo should only navigate to home when logged in

File: packages/visitorWeb/src/components/Header/Header.tsx [hunk 6]
```diff
@@ -97,9 +91,30 @@ const Header = () => {
...
             )}
+            <TranslateIcon fontSize="medium" />
           </a>
         </div>
       </div>
+      <Drawer anchor="left" open={showNavigationDrawer} onClose={() => setShowNavigationDrawer(false)} classes={{ paper: styles.drawer }}>
+        <div className={styles.drawerContent}>
+          <span className={styles.NavTitle}>
+            { loggedIn ? (
+              <a onClick={logoutUser}>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/Header/Header.tsx:102]: please dont use navigate() since the navbar doesnt automatically close after redirecting like that

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

File: packages/visitorWeb/src/components/Header/Header.tsx [hunk 7]
```diff
@@ -97,9 +96,31 @@ const Header = () => {
...
+                {t('components.Header.logout')}
+              </a>
+            ) : (
+              <a onClick={() => NavigateTo('/login', navigate, {})}>
```
Comment (JosefineM2000) [inline packages/visitorWeb/src/components/Header/Header.tsx:111]: This one still doesnt close after redirect

**Review by JosefineM2000 — CHANGES_REQUESTED** [review]

**Review by JosefineM2000 — APPROVED** [review]
